WalletWasabi/Services/UpdateManager.cs

```diff
@@ -22,11 +23,35 @@ public class UpdateManager : IDisposable
 	private const byte MaxTries = 2;
 	private const string ReleaseURL = "https://api.github.com/repos/zkSNACKs/WalletWasabi/releases/latest";
 
-	public UpdateManager(string dataDir, bool downloadNewVersion, IHttpClient httpClient)
+	public UpdateManager(string dataDir, bool downloadNewVersion, IHttpClient httpClient, UpdateChecker updateChecker)
 	{
 		InstallerDir = Path.Combine(dataDir, "Installer");
 		HttpClient = httpClient;
-		DownloadNewVersion = downloadNewVersion;
+
+		// The feature is disable on linux at the moment because we install Wasabi Wallet as a Debian package.
+		DownloadNewVersion = downloadNewVersion && !RuntimeInformation.IsOSPlatform(OSPlatform.Linux);
+
+		UpdateChecker = updateChecker;
+		UpdateChecker.UpdateStatusChanged += UpdateChecker_UpdateStatusChangedAsync;
+	}
+
+	public event EventHandler<UpdateStatus>? UpdateAvailableToGet;
+
+	public string InstallerDir { get; }
+	private IHttpClient HttpClient { get; }
+
+	///<summary>Comes from config file. Decides Wasabi should download the new installer in the background or not.</summary>
+	private bool DownloadNewVersion { get; }
+
+	///<summary>Install new version on shutdown or not.</summary>
+	public bool DoUpdateOnClose { get; set; }
+
+	private UpdateChecker UpdateChecker { get; }
+	private CancellationToken CancellationToken { get; set; }
+
+	public void Initialize(CancellationToken cancelationToken)
```

Comment:
I tried to get rid of this function and failed miserably. (unless I make some questionable changes)

Do you know why we need **THE** application stopping cancel token here?

Wouldn't a local CTS be enough? 
We could cancel and dispose the local CTS when the `Dispose()` method is called on stopping WW.
We could create that local CTS in `ctor`, then we don't need this method anymore.

WDYT?